docs/karma.md

```diff
@@ -0,0 +1,357 @@
+                        },
+                        {
+                            "name": "oauth",
+                            "reward": "3"
+                        },
+                        {
+                            "name": "token",
+                            "reward": "4"
+                        }
+                    ],
+                    "users": [
+                        {
+                            "user": {
+                                "chainId": "default",
+                                "local": "QjWhaN9qvpdI9MjS1YuL1GukwLc="
+                            },
+                            "sources": [
+                                {
+                                    "name": "oauth",
+                                    "count": "10"
+                                },
+                                {
+                                    "name": "token",
+                                    "count": "3"
+                                }
+                            ]
+                        }
+                    ],
+                }
+            }
+        }
+```
+This genesis file fragment will create three sources and give the user with local address
+`QjWhaN9qvpdI9MjS1YuL1GukwLc` 20 rewards from `oauth` and3 rewards from `token`.
+This user would than start with `20*10 + 3*4 = 42` karma.
+
+#### Users: UpdateSourcesForUser and DeleteSourcesForUser
+In a running DAppChain we can add a source to a user with `UpdateSourcesForUser`. We need a list of 
+name sof the new sources, plus a count of the number of rewards. The following code fragment
+gives an example of how you might do this, error checking skipped for readability.
+```go
+import `github.com/loomnetwork/go-loom/builtin/types/karma`
+
+func AddSourceForUser(name string, count int64, user, oracle loom.Addres, signer auth.Signer, karmaContact client.Contract) error {	
+	// Update the source information on the DAppChain
+	_, err := karmaContact.Call("UpdateSourcesForUser", &karma.KarmaStateUser{
+        User: user.MarshalPB(),
+        Oracle: oracle.MarshalPB(),
+        SourceStates: []*karma.KarmaSource{{name, count}},
+    }, signer, nil)
+	return err
+}
+```
+Similarly we can remove a source from those associated with a user on an existing DAppChain using the 
+DeleteSourcesForUser karma method.
+```go
+import `github.com/loomnetwork/go-loom/builtin/types/karma`
+
+func RemoveSourceForUser(name string, user, oracle loom.Addres, signer auth.Signer, karmaContact client.Contract) error {	
+	// Update the source information on the DAppChain
+	_, err := karmaContact.Call("UpdateSourcesForUser", &karma.KarmaStateKeyUser{
+        User: user.MarshalPB(),
+        Oracle: oracle.MarshalPB(),
+        StateKeys: []string{name},
+    }, signer, nil)
+	return err
+}
+```   
+## Karma methods
+The following methods can be called by anyone.
+* ` GetSources(ctx contract.StaticContext, ko *types.Address) (*ktypes.KarmaSources, error)` 
```

Comment:
You probably once again want to give examples of how to call these methods on command line, and later on we can add javascript or Go examples.